src/Uuid/ModelUuid.php

```diff
@@ -63,21 +63,50 @@ public function id(): string
 		// generate a new ID (to be saved in the content file)
 		$id = static::generate();
 
+		// store the new UUID
+		$this->model = static::storeId($this->model, $id);
+
+		// update the Uri object
+		$this->uri->host($id);
+
+		return $id;
+	}
+
+	/**
+	 * Retrieves the ID string (UUID without scheme) for the model
+	 * from the content file, if it is already stored there
+	 *
+	 * @param \Kirby\Cms\ModelWithContent $model
+	 */
+	public static function retrieveId(Identifiable $model): string|null
+	{
+		return $model->content('default')->get('uuid')->value();
+	}
+
+	/**
+	 * Stores the UUID for the model and makes sure
+	 * to update the content file and content object cache
+	 *
+	 * @param \Kirby\Cms\ModelWithContent $model
+	 * @param string $id
+	 */
+	public static function storeId(Identifiable $model, string $id): Identifiable
```

Comment:
I feel like this method should be `protected`. Users shouldn't be able to store a new UUID because it would break references. Also I think it can be an instance method, then we won't need to pass `$model`.

Reply:
I adjusted the signature of the method to retrieveId. It could indeed be an instance method. I'm fine with both ways. But I think a public method is actually quite useful. Not necessarily for other users, but for us.

Reply:
I had another look and I think you are right. By converting it to a protected method, we have more control and can rethink this method later if needed. If we make it public it immediately becomes a "feature" and if we want to change it it would be a breaking change. I also just converted it to non-static.

Reply:
We could also make it `public` with `@internal` later if we ever need it somewhere else.